Marlin/src/pins/lpc1769/pins_BTT_SKR_E3_TURBO.h

```diff
@@ -0,0 +1,266 @@
+#endif
+
+//
+// Steppers
+//
+#define X_STEP_PIN                         P1_04
+#define X_DIR_PIN                          P1_08
+#define X_ENABLE_PIN                       P1_00
+#ifndef X_CS_PIN
+  #define X_CS_PIN                         P1_01
+#endif
+
+#define Y_STEP_PIN                         P1_14
+#define Y_DIR_PIN                          P1_15
+#define Y_ENABLE_PIN                       P1_09
+#ifndef Y_CS_PIN
+  #define Y_CS_PIN                         P1_10
+#endif
+
+#define Z_STEP_PIN                         P4_29
+#define Z_DIR_PIN                          P4_28
+#define Z_ENABLE_PIN                       P1_16
+#ifndef Z_CS_PIN
+  #define Z_CS_PIN                         P1_17
+#endif
+
+#define E0_STEP_PIN                        P2_06
+#define E0_DIR_PIN                         P2_07
+#define E0_ENABLE_PIN                      P0_04
+#ifndef E0_CS_PIN
+  #define E0_CS_PIN                        P0_05
+#endif
+
+#define E1_STEP_PIN                        P2_11
+#define E1_DIR_PIN                         P2_12
+#define E1_ENABLE_PIN                      P0_21
+#ifndef E1_CS_PIN
+  #define E1_CS_PIN                        P0_22
+#endif
+
+#if HAS_TMC_UART
+  /**
+   * TMC2208/TMC2209 stepper drivers
+   *
+   * Hardware serial communication ports.
+   * If undefined software serial is used according to the pins below
+   */
+
+  //
+  // Software serial
+  //
+  #define X_SERIAL_TX_PIN                  P1_01
+  #define X_SERIAL_RX_PIN                  P1_01
+
+  #define Y_SERIAL_TX_PIN                  P1_10
+  #define Y_SERIAL_RX_PIN                  P1_10
+
+  #define Z_SERIAL_TX_PIN                  P1_17
+  #define Z_SERIAL_RX_PIN                  P1_17
+
+  #define E0_SERIAL_TX_PIN                 P0_05
+  #define E0_SERIAL_RX_PIN                 P0_05
+
+  #define E1_SERIAL_TX_PIN                 P0_22
+  #define E1_SERIAL_RX_PIN                 P0_22
+
+  // Reduce baud rate to improve software serial reliability
+  #define TMC_BAUD_RATE                    19200
+#endif
+
+//
+// TMC Low Power Standby pins
+//
+#define X_STDBY_PIN                        P3_26
+#define Y_STDBY_PIN                        P3_25
+#define Z_STDBY_PIN                        P1_18
+#define E0_STDBY_PIN                       P1_19
+#define E1_STDBY_PIN                       P2_13
+
+//
+// Temperature Sensors
+//
+#define TEMP_0_PIN                         P0_24
+#define TEMP_1_PIN                         P0_23
+//#define TEMP_2_PIN                       P1_30   // Onboard thermistor
+#define TEMP_BED_PIN                       P0_25
+
+//
+// Heaters / Fans
+//
+#define HEATER_0_PIN                       P2_03  // EXTRUDER 0
+#define HEATER_1_PIN                       P2_04  // EXTRUDER 1
+#define HEATER_BED_PIN                     P2_05  // BED
+#define FAN_PIN                            P2_01
+#define FAN1_PIN                           P2_02
+
+//
+// USB connect control
+//
+//#define USB_CONNECT_PIN                  P2_09
```

Comment:
What is this used for? Is there a reason it is commented out?

Reply:
I don't know how it's used. It was included in the original pins sent to me and was commented out. It's also in some of the other BigTreeTech pins files.

Looking at the schematic, it should probably be `P0_29`
![image](https://user-images.githubusercontent.com/13375512/94181978-1f5bb300-fe55-11ea-9a07-2bee93a53ca3.png)

Reply:
Actually I think you want to delete this pin altogether.

P2_09 is the correct pin number. It isn't supposed to be the data line, but actually is used to switch a pull up resistor attached to the D+ line. Here's the actual circuit it controls (from the SKR 1.4 SCH):

![image](https://user-images.githubusercontent.com/20053467/94219161-01b13c80-fe9b-11ea-9217-73877eac955e.png)

This pin only seems used by the STM32F1 HAL, and a few lines of code toggle this as Marlin boots. This pin definition is not used by the LPC HAL/framework at all.

The **pin** is still used, but on LPC and STM32F4 processors it is handled internally and doesn't need to be defined in a pins file.

Reply:
> Actually I think you want to delete this pin altogether.

Too late now! I'll try to remember to go back and remove it in a future PR.